core/actions/declaration.ts

```diff
@@ -34,8 +34,19 @@ export class Declaration extends ActionBuilder<dataform.Declaration> {
 
   public session: Session;
 
-  constructor(session?: Session) {
+  constructor(session?: Session, config?: dataform.ActionConfig) {
     super(session);
+    if (!session || !config) {
```

Comment:
I see why you do this - a slightly cleaner / less error prone way would be to overload the constructor to have a 0 params form and a 2 params form where they are both required, e.g: https://stackoverflow.com/questions/12702548/constructor-overload-in-typescript

Reply:
This doesn't seem to work the way one would expect it to with type overloading.

How about now, where just the presence of config is checked?

![image](https://github.com/dataform-co/dataform/assets/13434377/f4b27702-5675-4559-8fdb-6819a0771b4e)